10_rest/nickolee/pkg/puppy/rest/rest_test.go

```diff
@@ -0,0 +1,205 @@
+package rest
```

Comment:
This file is too similar to @willshen8. `rest.go` was inevitably going to look similar due to you both using `chi`, but testing could go many different ways.

Suggestion: split the one `TestRestAPI` into four, one for each method (eg: `TestGetPuppy`). This will also simplify the test data since each test case doesn't need a method (you'll know what it is based on the function you are in).

You can also look at using `httptest.Server` instead of `Recorder`, see how that changes your tests.